dlog/commitment/src/commitment.rs

```diff
@@ -60,19 +60,16 @@ pub fn shift_scalar<F: PrimeField>(x: F) -> F {
 }
 
 impl<C: AffineCurve> PolyComm<C> {
-    pub fn multi_scalar_mul(com: &Vec<&PolyComm<C>>, elm: &Vec<C::ScalarField>) -> Self {
-        PolyComm::<C> {
+    pub fn multi_scalar_mul(com: &Vec<&PolyComm<C>>, elm: &Vec<C::ScalarField>) -> Option<Self> {
+        Some(PolyComm::<C> {
             shifted: {
                 if com.len() == 0 || elm.len() == 0 || com[0].shifted == None {
                     None
                 } else {
                     let points = com
                         .iter()
-                        .map(|c| {
-                            assert!(c.shifted.is_some());
-                            c.shifted.unwrap()
-                        })
-                        .collect::<Vec<_>>();
+                        .map(|c| { c.shifted })
+                        .collect::<Option<Vec<_>>>()?;
```

Comment:
What is this call to collect doing?

Reply:
It's using [this impl](https://doc.rust-lang.org/std/option/enum.Option.html#impl-FromIterator%3COption%3CA%3E%3E). It's basically returning `None` if any of the values is `None`, or `Some(_)` if they're all `Some`s.

Reply:
Shouldn't we return an Error here instead of an Option? It seems like you would never want this behavior to not end up being an error.